### ISSUE TITLE
upload coverage report in GitHub Actions workflow, and remove Circle CI action

### ISSUE THREAD

**Issue by grahamgower** [popsim-consortium/stdpopsim#779]
It's fairly time consuming to learn/understand any one CI system, so to have both Circle and GitHub Actions (GHA) is quite a maintenance burden. I think GHA is simpler though? (Although GHA is still annoyingly complicated). I admit, I haven't tried to grok Circle, so have a bias towards GHA. If we do the codecov upload with GHA, then we can remove Circle. To remove GHA, we'd have to add a bunch of new OS/Python platform combinations to Circle.

**Comment by grahamgower** [comment]
@jeromekelleher [wrote](https://github.com/popsim-consortium/stdpopsim/pull/777#issuecomment-790491549)
> One advantage of having Circle going as well is that we can run CI on pypi dependencies. That's a good argument for keeping it IMO.

I don't quite follow. Why can't we do this with GHA?

**Comment by jeromekelleher** [comment]
> I don't quite follow. Why can't we do this with GHA?

We could, to be fair. The GHA scripts are already super complicated though. I guess if we had a separate action file that just ran on linux, we could make it pretty simple?

**Comment by grahamgower** [comment]
Yeah, a separate action file is probably the way to go, just like for the docs.

**Comment by jeromekelleher** [comment]
Also building and caching latest SLiM?

**Comment by grahamgower** [comment]
CircleCI has started failing pretty regularly, so now might be a good time to remove it.
